hazelcast-jet-core/src/test/java/com/hazelcast/jet/rocksdb/RocksMapTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.rocksdb;
+
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.jet.JetException;
+import com.hazelcast.jet.core.JetTestSupport;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static java.util.Collections.emptyMap;
+
+public class RocksMapTest extends JetTestSupport {
+    private RocksDBStateBackend rocksDBStateBackend;
+    private InternalSerializationService serializationService;
+    private RocksMap<String, Integer> rocksMap;
+
+
+    @Before
+    public void init() {
+        serializationService = getJetService(createJetMember())
+                .createSerializationService(emptyMap());
+        rocksDBStateBackend = new RocksDBStateBackend(serializationService);
+        rocksMap = rocksDBStateBackend.getMap();
+    }
+
+    @After
+    public void cleanup() {
+        rocksDBStateBackend.deleteKeyValueStore();
+        serializationService.dispose();
+    }
+
+
+    @Test
+    public void test_getMap() {
+        assert rocksMap != null;
+    }
+
+    @Test
+    public void test_put_and_get() {
+        rocksMap.put("Hello", 1);
+        assert rocksMap.get("Hello") == 1;
+    }
+
+    @Test
+    public void test_putAll_and_get() {
+        Map<String, Integer> map = new HashMap<>();
+        map.put("hello", 1);
+        map.put("bye", 2);
+        rocksMap.putAll(map);
+        assert rocksMap.get("bye") == 2;
+        assert rocksMap.get("hello") == 1;
+    }
+
+    @Test
+    public void test_putAll_and_getAll() {
```

Comment:
As the complexity of the test grows, you can see the advantage of adopting this structure:

```
// Given
... everything before the one thing that is under the test

// When
rocksMap.putAll(map);

// Then
... everything after that method call, including the assertions
```

This test isn't a unit test because it tries to test both `putAll` and `getAll` in one go. Tests should isolate individual functionalities.